src/modules/peek/Peek.FilePreviewer/Previewers/UnsupportedFilePreviewer/UnsupportedFilePreviewer.cs

```diff
@@ -22,69 +22,74 @@
 
         [ObservableProperty]
-        private UnsupportedFilePreviewData preview = new UnsupportedFilePreviewData();
+        private UnsupportedFilePreviewData preview = new();
 
         [ObservableProperty]
         private PreviewState state;
 
         public UnsupportedFilePreviewer(IFileSystemItem file)
         {
-            _folderSizeDispatcherTimer.Interval = TimeSpan.FromMilliseconds(500);
-            _folderSizeDispatcherTimer.Tick += FolderSizeDispatcherTimer_Tick;
-
             Item = file;
-            Preview.FileName = file.Name;
-            Preview.DateModified = file.DateModified?.ToString(CultureInfo.CurrentCulture);
             Dispatcher = DispatcherQueue.GetForCurrentThread();
         }
 
         private IFileSystemItem Item { get; }
 
         private DispatcherQueue Dispatcher { get; }
 
-        private Task<bool>? IconPreviewTask { get; set; }
-
-        private Task<bool>? DisplayInfoTask { get; set; }
-
-        public void Dispose()
-        {
-            _folderSizeDispatcherTimer.Tick -= FolderSizeDispatcherTimer_Tick;
-            GC.SuppressFinalize(this);
-        }
-
-        public Task<PreviewSize> GetPreviewSizeAsync(CancellationToken cancellationToken)
-        {
-            Size? size = new Size(680, 500);
-            var previewSize = new PreviewSize { MonitorSize = size, UseEffectivePixels = true };
-            return Task.FromResult(previewSize);
-        }
+        public Task<PreviewSize> GetPreviewSizeAsync(CancellationToken cancellationToken) =>
+            Task.FromResult(new PreviewSize { MonitorSize = new Size(680, 500), UseEffectivePixels = true });
 
         public async Task LoadPreviewAsync(CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            try
+            {
+                await Dispatcher.RunOnUiThread(async () =>
+                {
+                    Preview.FileName = Item.Name;
+                    Preview.DateModified = Item.DateModified?.ToString(CultureInfo.CurrentCulture);
 
-            State = PreviewState.Loading;
+                    State = PreviewState.Loaded;
 
-            IconPreviewTask = LoadIconPreviewAsync(cancellationToken);
-            DisplayInfoTask = LoadDisplayInfoAsync(cancellationToken);
+                    await LoadIconPreviewAsync(cancellationToken);
+                });
 
-            await Task.WhenAll(IconPreviewTask, DisplayInfoTask);
+                var progress = new Progress<string?>(update =>
+                {
+                    Dispatcher.TryEnqueue(() =>
+                    {
+                        Preview.FileSize = update;
+                    });
+                });
 
-            if (HasFailedLoadingPreview())
+                await LoadDisplayInfoAsync(progress, cancellationToken);
+            }
+            catch (OperationCanceledException)
             {
-                State = PreviewState.Error;
+                throw;
```

Comment:
```suggestion
```
Let's avoid rethrow this since it's a cancellation.

Reply:
Absolutely. There's a catch in `FilePreview.xaml.cs` which seems to be there because capturing and logging these exceptions was being considered:

https://github.com/microsoft/PowerToys/blob/dca8b7ac3560a77a57202398dd7e1e68e6eb9006/src/modules/peek/Peek.FilePreviewer/FilePreview.xaml.cs#L214C16-L218C37

Is that TODO no longer relevant? I assume `OperationCanceledException`s are quite common because users can quickly navigate to another file before the current one has finished previewing.

Reply:
Probably not relevant anymore. I don't think we want to have too many noisy `OperationCanceledExceptions` is the log file.